workflow/util/util.go

```diff
@@ -811,7 +811,9 @@ func FormulateRetryWorkflow(ctx context.Context, wf *wfv1.Workflow, restartSucce
 	switch wf.Status.Phase {
 	case wfv1.WorkflowFailed, wfv1.WorkflowError:
 	default:
```

Comment:
Would it make sense to pop this new code under a `case wfv1.WorkflowSucceeded:` as that's the intended and tested use case.

Reply:
done!